### PR TITLE
New user info cookie & unit test

### PR DESCRIPTION
**Pull request by Mayur-Khante**
[1] Added user info cookie holding user information.
[2] Added / Updated unit test.

### CODE REVIEW

File: internal/server.go [hunk 1]
```diff
@@ -120,47 +122,59 @@ func (s *Server) AuthCallbackHandler() http.HandlerFunc {
...
 		http.SetCookie(w, ClearCSRFCookie(r))
 
-		// Exchange code for token
-		token, err := p.ExchangeCode(redirectUri(r), r.URL.Query().Get("code"))
+		b, err := httputil.DumpRequest(r, true)
 		if err != nil {
-			logger.Errorf("Code exchange failed with: %v", err)
-			http.Error(w, "Service unavailable", 503)
+			logger.Warnf("httputil.DumpRequest: %v", err)
+			http.Error(w, "Not authorized", http.StatusUnauthorized)
```
Comment (muly) [inline internal/server.go]: please change this to 5xx error instead of 4xx

File: internal/provider/oidc.go [hunk 2]
```diff
@@ -88,6 +87,71 @@ func (o *OIDC) ExchangeCode(redirectURI, code string) (string, error) {
...
+		return "", fmt.Errorf("access token endpoint post json unmarshal: %w", err)
+	}
+	if token.AccessToken == "" {
+		return "", fmt.Errorf("access token empty: %s", string(data))
+	}
+	return token.AccessToken, nil
+}
+
+func getUserInfo(APIResourceURI, accessToken string) (User, error) {
+
```
Comment (muly) [inline internal/provider/oidc.go]: please cleanup the empty line at the start of the functions

File: internal/provider/oidc.go [hunk 3]
```diff
@@ -58,13 +61,9 @@ func (o *OIDC) Setup() error {
...
 		ClientID: o.ClientID,
 	})
-	o.UserURL = &url.URL{
-		Scheme: "https",
-		Host:   "cloudsso-test.cisco.com",
-		Path:   "/idp/userinfo.openid",
-	}
 
 	return nil
+
```
Comment (muly) [inline internal/provider/oidc.go]: please remove the empty line at the end of the function

**Review by muly — CHANGES_REQUESTED** [review]
please address these minor changes

**Review by muly — APPROVED** [review]

**Review by RamuRChenchaiah — APPROVED** [review]
Conditional:
1. Error message shall be externalised
2. The debug logs to follow certain format

**Pull request merged**